src/features/ajax/aggregate/gql.js

```diff
@@ -0,0 +1,97 @@
+import { isTrueObject } from '../../../common/util/type-check'
+
+/**
+ * @typedef {object} GQLMetadata
+ * @property {string} operationName Name of the operation
+ * @property {string} operationType Type of the operation
+ * @property {string} operationFramework Framework responsible for the operation
+ */
+
+/**
+ *
+ * @param {object|string} body Ajax request body
+ * @param {string} query Ajax request query param string
+ * @returns {GQLMetadata | undefined}
+ */
+export function parseGQL ({ body, query } = {}) {
+  if (!body && !query) return
+  const gqlBody = parseBatchGQL(parseGQLContents(body))
+  if (gqlBody) return gqlBody
+  const gqlQuery = parseSingleGQL(parseGQLQueryString(query))
+  if (gqlQuery) return gqlQuery
```

Comment:
You could probably reduce the overall `try..catch` blocks in this module by just putting the body of this function in a `try..catch`. If any error is thrown from any of the other functions, parsing should be considered failed and undefined should be returned. It is also the top level function of this module and provides the most protection from errors in the future.